The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5902546652/job/16010784501) [(plain)](https://github.com/rust-lang/rust/commit/d982bad9da0442f5120920f4b18810b9a8bb4def/checks/16010784501/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=benschulz
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_44f4cb4c-1bfc-48e8-bcdc-6bac3bf5a40b
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=mpsc-drop
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_44f4cb4c-1bfc-48e8-bcdc-6bac3bf5a40b
GITHUB_REF=refs/pull/114965/merge
GITHUB_REF_NAME=114965/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=ebfc3a926dc58975758de51a4ca9b446395a465d
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_44f4cb4c-1bfc-48e8-bcdc-6bac3bf5a40b
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_44f4cb4c-1bfc-48e8-bcdc-6bac3bf5a40b
GITHUB_TRIGGERING_ACTOR=benschulz
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/114965/merge
GITHUB_WORKFLOW_SHA=ebfc3a926dc58975758de51a4ca9b446395a465d
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_18_X64=/opt/hostedtoolcache/go/1.18.10/x64
---
---- [ui] tests/ui/span/send-is-not-static-std-sync.rs stdout ----
diff of stderr:

76 LL |     }
77    |     - `z` dropped here while still borrowed
78 LL |
- LL |     tx.use_ref(); // (channels drop glue does not use `z` => needs explicit use)
+ LL |     tx.use_ref(); // (channels' drop glue does not use `z` => needs explicit use)
81 
82 error: aborting due to 6 previous errors


---
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/span/send-is-not-static-std-sync.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/span/send-is-not-static-std-sync" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/span/send-is-not-static-std-sync/auxiliary"
stdout: none
--- stderr -------------------------------
error[E0505]: cannot move out of `y` because it is borrowed
   |
LL |     let y = Box::new(1);
LL |     let y = Box::new(1);
   |         - binding `y` declared here
LL |     let lock = Mutex::new(&x);
LL |     *lock.lock().unwrap() = &*y;
   |                             --- borrow of `*y` occurs here
LL |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here
...
LL |         *lock.lock().unwrap() = &z;


error[E0597]: `z` does not live long enough
   |
   |
LL |         let z = 2;
   |             - binding `z` declared here
LL |         *lock.lock().unwrap() = &z;
   |                                 ^^ borrowed value does not live long enough
LL |     }
   |     - `z` dropped here while still borrowed
LL |     //~^^ ERROR `z` does not live long enough
LL |     lock.use_ref(); // (Mutex is #[may_dangle] so its dtor does not use `z` => needs explicit use)


error[E0505]: cannot move out of `y` because it is borrowed
   |
LL |     let y = Box::new(1);
LL |     let y = Box::new(1);
   |         - binding `y` declared here
LL |     let lock = RwLock::new(&x);
LL |     *lock.write().unwrap() = &*y;
   |                              --- borrow of `*y` occurs here
LL |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here
...
LL |         *lock.write().unwrap() = &z;


error[E0597]: `z` does not live long enough
   |
   |
LL |         let z = 2;
   |             - binding `z` declared here
LL |         *lock.write().unwrap() = &z;
   |                                  ^^ borrowed value does not live long enough
LL |     }
   |     - `z` dropped here while still borrowed
LL |     //~^^ ERROR `z` does not live long enough
LL |     lock.use_ref(); // (RwLock is #[may_dangle] so its dtor does not use `z` => needs explicit use)


error[E0505]: cannot move out of `y` because it is borrowed
   |
LL |     let y = Box::new(1);
LL |     let y = Box::new(1);
   |         - binding `y` declared here
...
LL |     tx.send(&*y);
   |             --- borrow of `*y` occurs here
LL |     drop(y); //~ ERROR cannot move out
   |          ^ move out of `y` occurs here
...
LL |         tx.send(&z).unwrap();


error[E0597]: `z` does not live long enough
   |
   |
LL |         let z = 2;
   |             - binding `z` declared here
LL |         tx.send(&z).unwrap();
   |                 ^^ borrowed value does not live long enough
LL |     }
   |     - `z` dropped here while still borrowed
LL |     //~^^ ERROR `z` does not live long enough
LL |     tx.use_ref(); // (channels' drop glue does not use `z` => needs explicit use)

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0505, E0597.
```

</details>

--- a/library/std/src/sync/mpsc/mod.rs
+++ b/library/std/src/sync/mpsc/mod.rs
@@ -626,11 +626,6 @@ impl<T> Clone for Sender<T> {
     }
 }
 
-#[stable(feature = "rust1", since = "1.0.0")]
-impl<T> Drop for Sender<T> {
-    fn drop(&mut self) {}
-}
-
 #[stable(feature = "mpsc_debug", since = "1.8.0")]
 impl<T> fmt::Debug for Sender<T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
@@ -753,11 +748,6 @@ impl<T> Clone for SyncSender<T> {
     fn clone(&self) -> SyncSender<T> {
         SyncSender { inner: self.inner.clone() }
     }
-}
-
-#[stable(feature = "rust1", since = "1.0.0")]
-impl<T> Drop for SyncSender<T> {
-    fn drop(&mut self) {}
 }
 
 #[stable(feature = "mpsc_debug", since = "1.8.0")]
@@ -1094,11 +1084,6 @@ impl<T> IntoIterator for Receiver<T> {
     fn into_iter(self) -> IntoIter<T> {
         IntoIter { rx: self }
     }
-}
-
-#[stable(feature = "rust1", since = "1.0.0")]
-impl<T> Drop for Receiver<T> {
-    fn drop(&mut self) {}
 }
 
 #[stable(feature = "mpsc_debug", since = "1.8.0")]

--- a/tests/ui/span/send-is-not-static-std-sync.rs
+++ b/tests/ui/span/send-is-not-static-std-sync.rs
@@ -46,7 +46,7 @@ fn channel() {
         tx.send(&z).unwrap();
     }
     //~^^ ERROR `z` does not live long enough
-    // (channels lack #[may_dangle], thus their dtors are implicit uses of `z`)
+    tx.use_ref(); // (channel drop glue does not use `z` => needs explicit use)
 }
 
 fn main() {}

--- a/tests/ui/span/send-is-not-static-std-sync.stderr
+++ b/tests/ui/span/send-is-not-static-std-sync.stderr
@@ -75,11 +75,9 @@ LL |         tx.send(&z).unwrap();
    |                 ^^ borrowed value does not live long enough
 LL |     }
    |     - `z` dropped here while still borrowed
-...
-LL | }
-   | - borrow might be used here, when `tx` is dropped and runs the `Drop` code for type `Sender`
-   |
-   = note: values in a scope are dropped in the opposite order they are defined
+LL |
+LL |     tx.use_ref(); // (channel drop glue does not use `z` => needs explicit use)
+   |     -- borrow later used here
 
 error: aborting due to 6 previous errors
 

